src/components/TreeFilter/VirtualizedTreeView.tsx

```diff
@@ -77,21 +78,41 @@ export class VirtualizedTreeView extends React.PureComponent<IVirtualizedTreeVie
             });
             if (scrollToIndex === -1) {
                 // find the top level id of the nested checked item
-                let parent = this.parentLookup[checkedItemIds[0]];   
+                let parent = this.parentLookup[checkedItemIds[0]];
                 while (parent !== undefined) {
                     if (typeof parent.id === 'number') {
                         scrollToIndex = parent.id;
                         break;
                     }
                     parent = this.parentLookup[parent.id];
-                }          
+                }
             }
             this.setState({ scrollToIndex });
         }
     }
 
     public componentWillReceiveProps(nextProps: IVirtualizedTreeViewProps) {
-        if (nextProps.items !== this.props.items) {
+        const newItems = [];
+        if (this.state.searchedItemList !== null) {
+            this.state.searchedItemList.map((data) => {
+                const item = nextProps.items.find(x => x.id === data.id);
+                if (item) {
+                    newItems.push(item);
+                }
+            });
+
+            if (newItems.length !== 0 && newItems !== this.props.items) {
+                const filteredItems = ItemOperator.filterItems(newItems, this.state.searchText);

```

Comment:
why can't we just do` const filteredItems = ItemOperator.filterItems(nextProps.items` as it was before?

Reply:
Because when we search something we want only searched items until we clear search field and nextProps.items is list of all items.